test/spec/modules/teadsBidAdapter_spec.js

```diff
@@ -441,7 +449,129 @@ describe('teadsBidAdapter', () => {
+          const bidRequest = {
+            ...baseBidRequest,
+            userId: {
+              flocId: {
+                id: 'my-floc-id'
+              }
+            }
+          };
+
+          const request = spec.buildRequests([bidRequest], bidderResquestDefault);
+          const payload = JSON.parse(request.data);
+
+          expect(payload.cohortId).to.equal('my-floc-id');
+          expect(payload).not.to.have.property('cohortVersion');
+        });
+
+        it('should add cohortId and cohortVersion params to payload if FLoC ID system is enabled', function () {
+          const bidRequest = {
+            ...baseBidRequest,
+            userId: {
+              flocId: {
+                id: 'my-floc-id',
+                version: 'chrome.1.1'
+              }
+            }
+          };
+
+          const request = spec.buildRequests([bidRequest], bidderResquestDefault);
+          const payload = JSON.parse(request.data);
+
+          expect(payload.cohortId).to.equal('my-floc-id');
+          expect(payload.cohortVersion).to.equal('chrome.1.1');
+        });
+      });
+
+      describe('Unified ID v2', function () {
+        it('should not add unifiedId2 param to payload if uid2 system is not enabled', function () {
+          const bidRequest = {
+            ...baseBidRequest,
+            userId: {} // no "uid2" property -> assumption that the Unified ID v2 system is disabled
+          };
+
+          const request = spec.buildRequests([bidRequest], bidderResquestDefault);
+          const payload = JSON.parse(request.data);
+
+          expect(payload).not.to.have.property('unifiedId2');
+        });
+
+        it('should add unifiedId2 param to payload if uid2 system is enabled', function () {
+          const bidRequest = {
+            ...baseBidRequest,
+            userId: {
+              uid2: {
+                id: 'my-unified-id-2'
+              }
+            }
+          };
+
+          const request = spec.buildRequests([bidRequest], bidderResquestDefault);
+          const payload = JSON.parse(request.data);
+
+          expect(payload.unifiedId2).to.equal('my-unified-id-2');
+        })
+      });
+
+      describe('First-party cookie Teads ID', function () {
+        const storage = getStorageManager(132, 'teads');
+
+        afterEach(function () {
+          // drop cookie
+          storage.setCookie('_tfpvi', '', new Date(0));
+        });
+
+        it('should not add firstPartyCookieTeadsId param to payload if first-party cookie is not available', function () {
+          const request = spec.buildRequests([baseBidRequest], bidderResquestDefault);
+          const payload = JSON.parse(request.data);
+
+          expect(payload).not.to.have.property('firstPartyCookieTeadsId');
+        });
+
+        it('should add firstPartyCookieTeadsId param to payload if first-party cookie is available', function () {
+          storage.setCookie('_tfpvi', 'my-teads-id');
```

Comment:
We do not want to actually be setting and reading from storage / cookies inside of tests.

We should be mocking the stubbing / mocking the storage manager, which has its own tests to verify it works.

Here is an example:

You'll need to export your storage var so that your tests can mock it:
https://github.com/prebid/Prebid.js/blob/master/modules/rubiconAnalyticsAdapter.js#L11

Then in your beforeEach you should setup the stub
https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconAnalyticsAdapter_spec.js#L596-L600

Don't forget to restore it back in the afterEach
https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconAnalyticsAdapter_spec.js#L629-L631

Then in your tests you can set them up to return what you want and make sure your module works as expected:
https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconAnalyticsAdapter_spec.js#L1297